gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/OAuthUserInfoServiceHelper.java

```diff
@@ -180,9 +180,10 @@ <T extends OAuth2User> T getOAuthSpinnakerUser(T oAuth2User, OAuth2UserRequest u
               roles,
               username,
               oidcUser.getIdToken(),
-              oidcUser.getUserInfo());
-      spinnakerUser.getAttributes().putAll(details);
-      spinnakerUser.getAuthorities().addAll(oAuth2User.getAuthorities());
+              oidcUser.getUserInfo(),
+              details,
+              Optional.ofNullable(oidcUser.getAuthorities()).orElse(new ArrayList<>()).stream()
```

Comment:
```suggestion
              Optional.ofNullable(oidcUser.getAuthorities()).orElse(List.of());
```
here and below.

Reply:
but then it seems like the constructor handles null, so why not pass plain old `oidcUser.getAuthorities()` here?

Reply:
passing plain old oidcUser.getAuthorities()
https://github.com/spinnaker/gate/pull/1887/commits/2b67a513a18c7121e81510a3ed10573acb5603db